gobblin-service/src/test/java/org/apache/gobblin/service/modules/orchestration/DagProcessingEngineTest.java

```diff
@@ -80,17 +81,17 @@ public void setUp() throws Exception {
     doReturn(true).when(dagActionStore).deleteDagAction(any());
     dagManagementTaskStream =
         new DagManagementTaskStreamImpl(config, Optional.of(mock(DagActionStore.class)),
-            mock(MultiActiveLeaseArbiter.class), Optional.of(mock(DagActionReminderScheduler.class)), false,
-            dagManagementStateStore);
+            mock(MultiActiveLeaseArbiter.class), Optional.of(mock(DagActionReminderScheduler.class)),
+            false, dagManagementStateStore, Mockito.mock(DagProcessingEngineMetrics.class));
     this.dagProcFactory = new DagProcFactory(null);
 
     DagProcessingEngine.DagProcEngineThread dagProcEngineThread =
         new DagProcessingEngine.DagProcEngineThread(dagManagementTaskStream, this.dagProcFactory,
-            dagManagementStateStore, 0);
+            dagManagementStateStore, mock(DagProcessingEngineMetrics.class), 0);
```

Comment:
lots of tests are now being retrofitted to initialize w/ DPE metrics w/ a mock.

is it not possible to verify behavior or what's invoked on that mock?

Reply:
I'm not sure if we verify metrics being marked all that often in our tests. It could be checked although seems a bit overkill. I could use Mockito to verify that the specific metrics marking function is called (success/failed) for its corresponding case. Is that what you're imagining?